qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1090,4 +1100,64 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 		}
 		return resolvedJavaType;
 	}
-}
+
+	private void validateSwitchWhenPart(JavaMemberInfo member, Section ownerSection, String projectUri,
+			ResolvingJavaTypeContext resolvingJavaTypeContext, PropertyPart propertyPart, MethodPart methodPart,
+			List<Diagnostic> diagnostics) {
+		String javaTypeParent = member.getJavaElementType();
+		String partName = "";
+		if (methodPart != null) {
+			partName = methodPart.toString();
+		} else if (propertyPart != null) {
+			partName = propertyPart.toString();
+		}
+		List<Node> children = ownerSection.getChildren();
+		// Iterate through children and ensure the type of the #case or #is parts are
+		// the same as the parent section
+		for (Node node : children) {
+			if (node.getKind() == NodeKind.Section) {
+				List<Parameter> value = new ArrayList<Parameter>();

```

Comment:
Here I will define

Section childSection = (Section) node;

And I will check with childSection.getSectionKind if it is a  #is or#case

And you could call

value = childSection.getParameters()